trendmicrovisionone_connector.py

```diff
@@ -560,128 +378,210 @@ def artifact_exists(self, sdi, container):
+        """
+        Fetch container ID if it exists.
+        Args:
+            alert (Union[SaeAlert, TiAlert]): SaeAlert or TiAlert.
+        Raises:
+            RuntimeError: Raise an error if REST call fails.
+        Returns:
+            ID: Return the container ID.
+        """
         # check if TM workbenchID already exists in Splunk
         url = f'{self.get_phantom_base_url()}rest/container?_filter_source_data_identifier="{sdi}"&_filter_asset={self.get_asset_id()}'
-        # make rest call
+        # Make rest call
         try:
             response = requests.get(url, verify=False, timeout=30)  # nosemgrep
-        except Exception:
-            return None
+        except Exception as e:
+            raise RuntimeError(
+                "Encountered an error getting the existing container ID from Phantom."
+            ) from e
+
         # return id or None
-        if response.json().get("data", None):
-            return response.json().get("data", None)[0].get("id", None)
-        else:
+        container_data: dict[str, Any] = response.json()
+        if "data" not in container_data or len(container_data["data"]) == 0:
             return None
+        # This direct access is okay because the values MUST exist otherwise the problem is out of scope.
+        return container_data["data"][0]["id"]
 
-    def create_container(self, i, sdi):
-        # create a new container for a Trend Micro WorkBench incident
-        container = {}
-        container["name"] = i["workbenchName"]
-        container["source_data_identifier"] = sdi
-        container["label"] = self.get_config().get("ingest", {}).get("container_label")
-        container["description"] = i["detail"].get("description", "")
-        container["data"] = i
-        container["type"] = i["detail"].get("alertProvider", "")
-        container["severity"] = i.get("severity", "")
-        container["start_time"] = i.get("createdTime", "")
+    def _get_existing_container_id_for_alert(
+        self, alert: Union[SaeAlert, TiAlert]
+    ) -> Optional[int]:
+        """
+        Fetch container ID if it exists.
+        Args:
+            alert (Union[SaeAlert, TiAlert]): SaeAlert or TiAlert.
+        Raises:
+            RuntimeError: Raise an error if REST call fails.
+        Returns:
+            ID: Return the container ID.
+        """
+        return self._get_existing_container_id_for_sdi(alert.id)
+
+    def _create_new_container_payload(
+        self, alert: Union[SaeAlert, TiAlert]
+    ) -> Dict[str, Any]:
+        """
+        Returns information for an Alert
+        Args:
+            alert (Union[SaeAlert, TiAlert]): SaeAlert or TiAlert object.
+        Returns:
+            Dict[str, Any]: All pertinent data used to create container from Alert.
+        """
+        return {
+            "name": alert.model,
+            "source_data_identifier": alert.id,
+            "label": self.get_config().get("ingest", {}).get("container_label"),
```

Comment:
Because the self.config is loaded during the initialize() call, you can just reference self.config rather than loading it with self.get_config()